### PR TITLE
pkg/filenotify/poller fixes

### PR DESCRIPTION
**Pull request by kolyshkin**
A couple of fixes to issues I found while reading the code. Might help reduce the number of mysterious bugs on Windows, where poller is used for logs. See individual commits for descriptions.

three-bugs-with-green-umbrellas-hd-insects-fun-wallpaper

### CODE REVIEW

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by tiborvass — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
This bug was revealed because since https://github.com/moby/moby/pull/37412, this code is used for Windows logging

**Review by AntaresS — APPROVED** [review]
LGTM

**Comment by kolyshkin** [comment]
> This bug
Actually three (as depicted :smiley_cat:)

File: pkg/filenotify/poller.go [hunk 1]
```diff
@@ -146,12 +147,11 @@ func (w *filePoller) sendErr(e error, chClose <-chan struct{}) error {
 func (w *filePoller) watch(f *os.File, lastFi os.FileInfo, chClose chan struct{}) {
 	defer f.Close()
 	for {
-		time.Sleep(watchWaitTime)
 		select {
+		case <-time.After(watchWaitTime):
```
Comment (LK4D4) [inline pkg/filenotify/poller.go]: You might have stray timers because of `After`, it's better to use `NewTimer` + `Stop` later I think.

Comment (tiborvass) [reply pkg/filenotify/poller.go]: you mean Stop in defer ?

Comment (kolyshkin) [reply pkg/filenotify/poller.go]: Will the timer still exit? in here the watchWaitTime is 200ms

Comment (cpuguy83) [reply pkg/filenotify/poller.go]: Agree this should be fine.

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**